src/components/chat-view/Chat.tsx

```diff
@@ -275,7 +293,18 @@ const Chat = forwardRef<ChatRef, ChatProps>((props, ref) => {
           setChatMessages((prevChatHistory) =>
             prevChatHistory.map((message) =>
               message.role === 'assistant' && message.id === responseMessageId
-                ? { ...message, content: message.content + content }
+                ? {
+                    ...message,
+                    content: message.content + content,
+                    metadata: {
+                      ...message.metadata,
+                      usage: sumTokenUsages([
+                        message.metadata?.usage,
+                        chunk.usage,
+                      ]),
+                      model: chatModel,
+                    },
+                  }
```

Comment:
It would be better to aggregate token usage from the LLM provider instead of here.